compiler/src/dmd/root/array.d

```diff
@@ -55,6 +55,19 @@ public:
         if (data.ptr && data.ptr != &smallarray[0])
             mem.xfree(data.ptr);
     }
+
+
+    // this is a "dummy" template because of c++ header generation
+    // challenges with wrapping this in static if instead
+    extern(D) this()(T[] elems ...) pure nothrow if (is(T == struct) || is(T == class))
```

Comment:
```suggestion
    // this is using a template constraint because of ambiguity with this(size_t) when T is
    // int, and c++ header generation doesn't accept wrapping this in static if
    extern(D) this()(T[] elems ...) pure nothrow if (is(T == struct) || is(T == class))
```
Just to clarify there are two reasons for this.